wgsl/wgsl_spec_style_guide.md

```diff
@@ -0,0 +1,79 @@
+# WSGL spec writing style guide
+
+Goal:  Avoid possibly being misunderstood.
+*   Tradeoff: The text might read as stilted.  **Precision is better than flair.**
+
+Prefer short sentences.
+
+Be clear about which agent is doing what:
+*   The _shader author_ is the person writing the WGSL code for use in their application
+*   The _implementation_ is the thing that runs the application, processing the WGSL code.
+    *   Implicitly, the implementation is a hardware-software combination, and we don’t care about the boundary between them.
```

Comment:
This may be a little too aspirational?
We want to not distinguish between them, but sometimes it's useful to explain (in non-normative text), what the HW is doing, so that it explains the API design.

Reply:
I'll add a note that this is really about the normative part of the text.